python/modules/EventInfo.py

```diff
@@ -13,35 +13,56 @@
 class EventInfo(Module):
     def __init__(
         self,
-        storeVariables = []
+        storeVariables = [],
+        useExistingSum = False
     ):
         self.storeVariables = storeVariables
+        self.useExistingSum = useExistingSum
 
     def beginJob(self):
         pass
     def endJob(self):
         pass
     def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
         self.nGenWeights = 0
+        self.nGenEventCount = 0
+        self.genEventSumw2 = 0
+        self.LHEScaleSumw = [0, 0, 0, 0, 0, 0, 0, 0, 0]
+        self.existingSumSaved = False
         self.out = wrappedOutputTree
         for variable in self.storeVariables:
             variable[0](self.out)
         
     def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
     	if not Module.globalOptions["isData"]:
+            nGenEventCount_parameter = ROOT.TParameter(float)("genEventCount", self.nGenWeights)
             nGenWeight_parameter = ROOT.TParameter(float)("sumGenWeights", self.nGenWeights)
+            genEventSumw2_parameter = ROOT.TParameter(float)("sumGenWeights2", self.genEventSumw2)
+            LHEScaleSumw_parameter = []
+            for i in range(9): LHEScaleSumw_parameter.append(ROOT.TParameter(float)("LHEScaleSumw_{}".format(i), self.LHEScaleSumw[i]/self.nGenWeights))
+
             outputFile.cd()
+            nGenEventCount_parameter.Write()
             nGenWeight_parameter.Write() 
+            genEventSumw2_parameter.Write()
+            for i in range(9): LHEScaleSumw_parameter[i].Write()
         else:
             pass
         
     def analyze(self, event):
 
         if not Module.globalOptions["isData"]:
-            if hasattr(event, 'genEventSumw'):
-                self.nGenWeights += event.genEventSumw
+            if self.useExistingSum and (not self.existingSumSaved):
+                self.nGenWeights = event.genEventSumw
```

Comment:
unfortunately this won't work for Run 3 actually because genEventSumw is not saved in the Events branch but in the Runs one....here an example. so either we try a way to get it from the correct branch or we could just continue to manually calculated the sum
![Screenshot from 2024-03-26 18-45-26](https://github.com/gmilella12/nanoAOD-tools/assets/68775397/f1ed8481-16b6-4f24-b2b0-cec10e521fb1)

Reply:
fastest solution could be just to use the "manually' sum method. so, could you get rid of the flag regarding the existingSum and similia and make a new commit? then, for the rest i think it is all fine